tests/acceptance-tests/src/acceptance/rds_broker_test.go

```diff
@@ -1,22 +1,101 @@
+	"os/exec"
+	"time"
+
 	"github.com/cloudfoundry-incubator/cf-test-helpers/cf"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/generator"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/helpers"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gbytes"
 	. "github.com/onsi/gomega/gexec"
 )
 
-var (
-	brokerName = "postgres"
-)
-
 var _ = Describe("RDS broker", func() {
+	const (
+		brokerName   = "postgres"
+		testPlanName = "temporary-test-plan"
+	)
 
 	It("should be registered", func() {
 		plans := cf.Cf("marketplace").Wait(DEFAULT_TIMEOUT)
 		Expect(plans).To(Exit(0))
 		Expect(plans).To(Say(brokerName))
 	})
+
+	Context("creating a database instance", func() {
+		// Avoid creating additional tests in this block because this setup and teardown is
+		// slow (several minutes).
+
+		const (
+			DB_CREATE_TIMEOUT = 15 * time.Minute
+		)
+
+		var (
+			appName        string
+			dbInstanceName string
+		)
+		BeforeEach(func() {
+			appName = generator.PrefixedRandomName("CATS-APP-")
+			dbInstanceName = generator.PrefixedRandomName("test-db-")
+			Expect(cf.Cf("create-service", brokerName, testPlanName, dbInstanceName).Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			fmt.Fprint(GinkgoWriter, "Polling for RDS creation to complete")
+			Eventually(func() *Buffer {
+				fmt.Fprint(GinkgoWriter, ".")
+				command := quietCf("cf", "service", dbInstanceName).Wait(DEFAULT_TIMEOUT)
+				Expect(command).To(Exit(0))
+				return command.Out
+			}, DB_CREATE_TIMEOUT, 15*time.Second).Should(Say("create succeeded"))
+			fmt.Fprint(GinkgoWriter, "done\n")
+		})
+
+		AfterEach(func() {
+			cf.Cf("delete", appName, "-f").Wait(DEFAULT_TIMEOUT)
+
+			Expect(cf.Cf("delete-service", dbInstanceName, "-f").Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+			// Poll until destruction is complete, otherwise the org cleanup fails.
+			fmt.Fprint(GinkgoWriter, "Polling for RDS destruction to complete")
+			Eventually(func() *Buffer {
+				fmt.Fprint(GinkgoWriter, ".")
+				command := quietCf("cf", "services").Wait(DEFAULT_TIMEOUT)
+				Expect(command).To(Exit(0))
+				return command.Out
+			}, DB_CREATE_TIMEOUT, 15*time.Second).ShouldNot(Say(dbInstanceName))
+			fmt.Fprint(GinkgoWriter, "done\n")
+		})
+
+		It("can be used from an app", func() {
+			Expect(cf.Cf(
+				"push", appName,
+				"--no-start",
+				"-b", config.GoBuildpackName,
+				"-p", "../../../example-apps/db-healthcheck",
+				"-f", "../../../example-apps/db-healthcheck/manifest.yml",
+				"-d", config.AppsDomain,
+			).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+
+			Expect(cf.Cf("bind-service", appName, dbInstanceName).Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			Expect(cf.Cf("start", appName).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+
+			// Hit app endpoint to verify functionality.
+			fmt.Fprintln(GinkgoWriter, "Sending request to DB Healthcheck app")
+			resp, err := httpClient.Get(helpers.AppRootUri(appName))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.StatusCode).To(Equal(200))
```

Comment:
Is the db-healthcheck application written in such a way that a 200 status code definitely means the database is up an running? Is there any output to the page we should be asserting on?